src/server/system_services/account_server.js

```diff
@@ -719,6 +719,11 @@ async function add_external_connection(req) {
     info.secret_key = system_store.master_key_manager.encrypt_sensitive_string_with_master_key_id(
         req.rpc_params.secret, req.account.master_key_id._id);
 
+    info.azure_tenant_id = req.rpc_params.azure_log_access_keys.azure_tenant_id || undefined;
+    info.azure_client_id = req.rpc_params.azure_log_access_keys.azure_client_id || undefined;
+    info.azure_client_secret = req.rpc_params.azure_log_access_keys.azure_client_secret || undefined;
+    info.azure_logs_analytics_workspace_id = req.rpc_params.azure_log_access_keys.azure_logs_analytics_workspace_id || undefined;
```

Comment:
Maybe change to optional chaining (?.)
```suggestion
    info.azure_tenant_id = req.rpc_params.azure_log_access_keys?.azure_tenant_id;
    info.azure_client_id = req.rpc_params.azure_log_access_keys?.azure_client_id;
    info.azure_client_secret = req.rpc_params.azure_log_access_keys?.azure_client_secret;
    info.azure_logs_analytics_workspace_id = req.rpc_params.azure_log_access_keys?.azure_logs_analytics_workspace_id;
```